db/internal_stats.cc

```diff
@@ -1454,6 +1605,15 @@ void InternalStats::DumpCFStatsNoFileHistogram(std::string* value) {
   cf_stats_snapshot_.ingest_keys_addfile = ingest_keys_addfile;
   cf_stats_snapshot_.comp_stats = compaction_stats_sum;
   cf_stats_snapshot_.stall_count = total_stall_count;
+
+  Status s = CollectCacheEntryStats();
```

Comment:
So we're dumping the same thing for each CF here, right? (I realize we have no way of knowing which CF or in fact which DB a certain cache entry came from.)

Reply:
Yes, but I include an "id" for the block cache, including its pointer location, so that you can de-duplicate manually or even programmatically from GetMapProperty data for various CFs. I might soon implement this de-duplication internally for reporting within FB infrastructure.

I thought about the id including process id also, for guaranteed uniqueness within a host, but didn't go that far since we don't appear to have a general API for that.

And the age-based result caching should prevent excessive scans. :)